meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/factory/ZkMetaClientFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.helix.metaclient.impl.zk.factory;
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+import org.apache.helix.metaclient.factories.MetaClientConfig;
+import org.apache.helix.metaclient.factories.MetaClientFactory;
+import org.apache.helix.metaclient.impl.zk.ZkMetaClient;
+
+
+public class ZkMetaClientFactory extends MetaClientFactory {
+  @Override
+  public MetaClientInterface getMetaClient(MetaClientConfig config) {
+    if (config.getStoreType() == MetaClientConfig.StoreType.ZOOKEEPER
+        && config instanceof ZkMetaClientConfig) {
+      return new ZkMetaClient((ZkMetaClientConfig) config);
+    }
+    return null;
```

Comment:
I'm thinking if we should just throw `IllegalArgumentException` instead of return null. What's your thought on this?